docs/rfcs/004-session-management.md

```diff
@@ -0,0 +1,125 @@
+
+* Unsaved buffer changes;
+* Per-file manually changed indentation, line ending, and encoding settings.
+
+### Add an explicit indication of the current Session to the tree-view
+
+At the top of the tree-view, display the current Project's name. Clicking it opens the project management UI described below.
+
+> TODO: graphic
+
+### Add a command-line argument to explicitly choose the Session
+
+When opening Atom from the command line with one or more locations specified:
+
+1. If exactly one non-open Session matches those paths, that Session is loaded into the newly opened Atom window.
+2. If no non-open Sessions match, a new Session is created for that Atom window.
+3. If multiple Sessions match, a new Session is created for that Atom window, and a notification is displayed allowing the user to switch to any of the matching Sessions and destroy the just-created one.
+
+In this context, "matches" means _every path in the set, normalized through realpath and case sensitivity, begins with the path of at least one project folder in the Session under consideration_, and, as a special case, _a Session containing zero project folders matches everything_.
+
+The `--session` or `-s` flags to the `atom` command-line script explicitly designate the name of a Session which the newly opened window should open. If no Session with that name exists, a new one is created. If a window is already open on the named Session, the provided paths are added to the existing window and it is brought to the foreground.
+
+```sh
+# Open a new Atom window containing:
+# * the root folder "src/project-one"
+# * TextEditor on "${HOME}/writing/tps-report.txt"
+$ atom --session work src/project-one ~/writing/tps-report.txt
+
+# Open a new Atom window containing:
+# * the root folder "src/game"
+$ atom -s fun src/game
+
+# Add the root folder "src/project-three" to the existing "work" window
+$ atom -s work src/project-three
+
+# (close the "work" window)
+
+# Re-open a new Atom window with:
+# * root folders "src/project-one" and "src/project-three"
+# * TextEditor on "${HOME}/writing/tps-report.txt"
+$ atom -s work
+```
+
+`--session` is mutually exclusive with the `--add` option, which always adds the associated paths to the matching or last-opened window's Session.
+
+### Create a new bundled package to list and manage serialized state
+
+Clicking the "Session" tile in the tree-view opens a workspace item for listing, creating, loading, modifying, and deleting known Sessions. This is where Sessions can be renamed and managed.
+
+> TODO: graphic
+
+Loading a different Session immediately serializes the window's state into the previously active project and deserializes the window state from the chosen one into this window.
+
+A separate tab within the Session management shows known unsaved buffers. This acts as a kind of "dead letter office" to track down and recover lost changes.
```

Comment:
☝️ Unsure about this being a "tab". Maybe just a button at the top that opens a different "lost and found" item? I'm kind of waiting to make some UI mocks to figure out the best way to do this.

Reply:
I'm a big fan of having a separate tree-view for sessions and the ability to manage them.